File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/board/domain/qna/answer/service/WriteAnswerService.java
package io.dustin.apps.board.domain.qna.answer.service;

import io.dustin.apps.board.domain.qna.answer.model.Answer;
import io.dustin.apps.board.domain.qna.answer.repository.AnswerRepository;
import io.dustin.apps.common.exception.DataNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.Optional;

@RequiredArgsConstructor
@Service
public class WriteAnswerService {
    private final AnswerRepository answerRepository;

    public Answer create(Long adminId, Long questionId, String content){

        Answer answer = Answer.builder()
                .content(content)
                .questionId(questionId)
                .adminId(adminId)
                .build();
        this.answerRepository.save(answer);
        return answer;
    }

    @Transactional
    public void updateContent(Answer answer, String content){
        answer.updateContent(content);
    }

    @Transactional
    public void delete(Answer answer){
        answer.delete();
    }

    public Answer findById(long answerId) {
        Optional<Answer> optional = this.answerRepository.findById(answerId);
        if(optional.isPresent()) {
            return optional.get();
        } else {
            return null;
        }
    }

    public Answer findByIdOrThrow(long answerId) {
        Optional<Answer> optional = this.answerRepository.findById(answerId);
        if(optional.isPresent()) {
            return optional.get();
        } else {
            throw new DataNotFoundException("""
                    id [#1]로 조회된 게시물이 없습니다.""".replace("#1", String.valueOf(answerId)));
        }
    }

}

File: apps/src/main/kotlin/io/dustin/apps/board/domain/qna/answer/service/WriteAnswerService.kt
package io.dustin.apps.board.domain.qna.answer.service

import io.dustin.apps.board.domain.qna.answer.model.Answer
import io.dustin.apps.board.domain.qna.answer.repository.AnswerRepository
import io.dustin.apps.common.exception.DataNotFoundException
import io.dustin.apps.common.repository.findByIdOrThrow
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service

@Service
class WriteAnswerService (
    private val answerRepository: AnswerRepository
) {
    /**
     * 답변 작성
     */
    fun create(adminId: Long, questionId: Long, content: String): Answer {
        val answer =  Answer(
            adminId = adminId,
            questionId = questionId,
            content = content
        )
        answerRepository.save(answer)
        return answer
    }

    /**
     * 답변 수정
     */
    fun updateContent(answer: Answer, content: String) {
        answer.updateContent(content)
    }

    /**
     * 답변 삭제
     */
    fun delete(answer: Answer) {
        answer.delete()
    }

    fun findByIdOrNull(answerId: Long): Answer? {
        return answerRepository.findByIdOrNull(answerId)
    }

    fun findByIdOrThrow(answerId: Long): Answer {
        return answerRepository.findByIdOrThrow(answerId)
    }
}
